src/core/symbology-ng/qgsrendererv2.h

```diff
@@ -238,7 +238,7 @@ class CORE_EXPORT QgsFeatureRendererV2
     Q_DECL_DEPRECATED virtual QDomElement writeSld( QDomDocument& doc, const QgsVectorLayer &layer ) const;
     //! create the SLD UserStyle element following the SLD v1.1 specs with the given name
     //! @note added in 2.8
-    virtual QDomElement writeSld( QDomDocument& doc, const QString& styleName ) const;
+    virtual QDomElement writeSld( QDomDocument& doc, const QString& styleName, QgsStringMap props = QgsStringMap() ) const;
```

Comment:
The main issue is here, you need to add a new method instead of adding a new parameter (it's virtual)